### PR TITLE
linking notifications to contest and messages

### PR DESCRIPTION
**Pull request by jeewanbhusal**
### What this PR does (required):
- This PR contains feature, which links to particular contests and message dashboard when notifications are clicked
- Added property contestId in the notification model , in order to achieve this feature
- Added some styling for notifications on-hover and also for notification badge

### Screenshots / Videos (required):
Screen Shot 2021-08-05 at 17 26 37
Screen Shot 2021-08-05 at 17 33 05

### Any information needed to test this feature (required):
- Go to notification box, to see your notifications. Simply click on any notification.

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: client/src/components/NotificationPopUp/NotificationPopUp.tsx [hunk 1]
```diff
@@ -115,9 +123,14 @@ const NotificationPopUp = (): JSX.Element => {
             </Link>
           </Grid>
         </Grid>
-        {notifications?.length
-          ? notifications?.map((notification) => (
-              <Grid direction="row" container key={notification._id}>
+        {notifications?.length ? (
+          notifications?.map((notification) => (
+            <Link
+              to={notification.contestId ? `/contest/${notification.contestId}` : `/dashboard`}
```
Comment (rajivtitus) [inline client/src/components/NotificationPopUp/NotificationPopUp.tsx:129]: 👍🏽

**Review by rajivtitus — APPROVED** [review]
LGTM!

File: client/src/components/NotificationPopUp/NotificationPopUp.tsx (hunk 1 above)
Comment (jeewanbhusal-1) [reply client/src/components/NotificationPopUp/NotificationPopUp.tsx:129]: thank you :)

**Comment by jeewanbhusal-1** [comment]
> LGTM!

great. thank you.

**Pull request merged**